plugin/central.vim

```diff
@@ -3,16 +3,43 @@
 " Version: 0.2.0
 " License: BSD
 
-set backupdir=~/.vim/backup//
-set directory=~/.vim/swap//
-set undodir=~/.vim/undo//
+set backupdir=$VIMHOME/backup//
+set directory=$VIMHOME/swap//
+set undodir=$VIMHOME/undo//
 
-if !isdirectory($HOME . "/.vim/backup")
-    call mkdir($HOME . "/.vim/backup", "p")
+for dir in [ &backupdir, &directory, &undodir ]
+    if !isdirectory(dir)
+        call mkdir(dir, 'p')
+    endif
+endfor
+
+set backup swapfile undofile
+
+if !exists('g:central_cleanup_enable')
+    let g:central_cleanup_enable = 30 "days
+endif
+
+if g:central_cleanup_enable > 0
+    let epoch = localtime() - (g:central_cleanup_enable * 86400)
+    for dir in [ &backupdir, &directory, &undodir ]
+        for file in split(glob(substitute(dir, '//', '/*', '')))
+            if getftime(file) < epoch
+                call delete(file)
+            endif
+        endfor
+    endfor
 endif
-if !isdirectory($HOME . "/.vim/swap")
-    call mkdir($HOME . "/.vim/swap", "p")
+
+if !exists('g:central_multiple_backup_enable')
+    let g:central_multiple_backup_enable = 1
 endif
-if !isdirectory($HOME . "/.vim/undo")
-    call mkdir($HOME . "/.vim/undo", "p")
+
+if g:central_multiple_backup_enable == 1
```

Comment:
This is actually a life saver.

I've had cases where, I thought I checked something into git, hadn't, checked that file out, and it was blown away :( 

Sweet addition ❤️

Reply:
Last week I managed to overwrite two days worth of work. By executing `:w` multiple times I lost my backup, and I also managed to screw up my undo history (not sure how I did the latter). Anyway, this is what prompted me to play around with multiple backups. The auto clean up feature was then a consequence of making sure the multiple backups did not get out of control.